ts/features/bonus/cgn/screens/merchants/CgnMerchantsListScreen.tsx

```diff
@@ -10,40 +11,84 @@ import BaseScreenComponent from "../../../../../components/screens/BaseScreenCom
 import { IOStyles } from "../../../../../components/core/variables/IOStyles";
 import I18n from "../../../../../i18n";
 import { emptyContextualHelp } from "../../../../../utils/emptyContextualHelp";
-import { availableMerchants } from "../../__mock__/availableMerchants";
 import { navigateToCgnMerchantDetail } from "../../navigation/actions";
 import CgnMerchantsListView from "../../components/merchants/CgnMerchantsListView";
 import { H1 } from "../../../../../components/core/typography/H1";
 import { IOColors } from "../../../../../components/core/variables/IOColors";
+import { Merchant } from "../../../../../../definitions/cgn/merchants/Merchant";
+import {
+  cgnOfflineMerchantsSelector,
+  cgnOnlineMerchantsSelector
+} from "../../store/reducers/merchants";
+import { OfflineMerchant } from "../../../../../../definitions/cgn/merchants/OfflineMerchant";
+import { OnlineMerchant } from "../../../../../../definitions/cgn/merchants/OnlineMerchant";
+import {
+  cgnOfflineMerchants,
+  cgnOnlineMerchants
+} from "../../store/actions/merchants";
+import { isLoading, isReady } from "../../../bpd/model/RemoteValue";
+import { LoadingErrorComponent } from "../../../bonusVacanze/components/loadingErrorScreen/LoadingErrorComponent";
 
 type Props = ReturnType<typeof mapStateToProps> &
   ReturnType<typeof mapDispatchToProps>;
 
-export type TmpMerchantType = {
-  name: string;
-  category: string;
-  location: string;
+const OFFLINE_FIXED_BOUNDINGBOX = {
+  userCoordinates: {
+    latitude: 41.827701462326985,
+    longitude: 12.66444625336996
+  },
+  boundingBox: {
+    coordinates: {
+      latitude: 34.845459548,
+      longitude: 6.5232427904
+    },
+    deltaLatitude: 6.9822419143,
+    deltaLongitude: 6.141203463
+  }
 };
 
 const DEBOUNCE_SEARCH: Millisecond = 300 as Millisecond;
+
+type MerchantsAll = OfflineMerchant | OnlineMerchant;
 /**
  * Screen that renders the list of the merchants which have an active discount for CGN
+ * This is the v1 of the merchants visualization for CGN discount, offline and online merchants are mixed together
  * @param props
  * @constructor
  */
 const CgnMerchantsListScreen: React.FunctionComponent<Props> = (
   props: Props
 ) => {
   const [searchValue, setSearchValue] = React.useState("");
-  const [merchantList, setMerchantsList] = React.useState(props.merchants);
+  const [merchantList, setMerchantsList] = React.useState<
+    ReadonlyArray<MerchantsAll>
+  >([]);
+
+  // Mixes online and offline merchants to render on the same list
+  // merchants are sorted by name
+  const merchantsAll = useMemo(() => {
+    const onlineMerchants = isReady(props.onlineMerchants)
+      ? props.onlineMerchants.value
+      : [];
+    const offlineMerchants = isReady(props.offlineMerchants)
+      ? props.offlineMerchants.value
+      : [];
+
+    return [
+      ...offlineMerchants,
+      ...onlineMerchants
+    ].sort((m1: MerchantsAll, m2: MerchantsAll) =>
+      m1.name > m2.name ? 1 : -1
```

Comment:
```suggestion
      m1.name.localeCompare(m2.name)
```
built-in `localeCompare` function could be used